### PR TITLE
Secret santa modifications and tests.

### PR DESCRIPTION
**Pull request by keithnewman**
- Modify `secret_santa()` so that it doesn't affect the seed for the user's session after the function has finished running.
- Add tests for the `secret_santa` function.
- Refactor duplicate entry check in `secret_santa()`.
- Stop people playing with 3 or fewer players since the players can determine who is gifting who in these scenarios (@jcken95)

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
-    email = "nrennie35@gmail.com", comment = c(ORCID = "0000-0003-4797-557X")))
-Description: R functions to help with christmas admin.
-Imports:
-  ggforce,
-  ggplot2,
-  sf
+Title: R functions to help with Christmas admin
+Version: 0.0.0.9002
+Authors@R: 
+    person("Nicola", "Rennie", , "nrennie35@gmail.com", role = c("aut", "cre", "cph"),
```
Comment (nrennie) [inline DESCRIPTION]: The formatting of Authors@R has gone a bit weird? Keep the args named, and avoid double comma.

File: R/secret_santa.R [hunk 2]
```diff
@@ -5,19 +5,30 @@
...
+  if (anyDuplicated(names)) {
     stop("Non-unique names provided - do you need to add a surname?")
   }
-  set.seed(seed)
-  random_names <- sample(names)
-  output <- data.frame(
-    gift_giver = random_names,
-    gift_recipient = random_names[c(2:length(random_names), 1)]
+
+  if (length(names) <= 3) {
```
Comment (nrennie) [inline R/secret_santa.R]: Feel like it should it be an error if 1 or 2 people, but a warning if 3? Might still want to use it for gift assignment even if it's not a secret.

File: DESCRIPTION (hunk 1 above)
Comment (nrennie) [reply DESCRIPTION]: Also, you've done quite a lot of work on this so feel like you should add yourself as an author 💯

**Review by nrennie — CHANGES_REQUESTED** [review]
Thank you @keithnewman (and @StatsRhian and @jcken95) - tiny changes requested 🎅

File: DESCRIPTION (hunk 1 above)
Comment (keithnewman) [reply DESCRIPTION]: strange... `usethis::use_tidy_description()` did that. I have undone it [edb24fc](https://github.com/nrennie/christmas/pull/1/commits/edb24fc9486ff2a1eee4f327cc38f48ce044b986)

Comment (keithnewman) [reply DESCRIPTION]: thanks, but you did all the work really.

File: R/secret_santa.R (hunk 2 above)
Comment (keithnewman) [reply R/secret_santa.R]: Sure thing https://github.com/nrennie/christmas/pull/1/commits/ae317f8baeb839b8492d3b1d764851a029e96a93

**Review by nrennie — APPROVED** [review]
Thank you!

**Pull request merged**